src/main/java/uk/gov/hmcts/ccd/data/casedetails/DefaultCaseDetailsRepository.java

```diff
@@ -76,8 +77,13 @@ public CaseDetails set(final CaseDetails caseDetails) {
         } catch (PersistenceException e) {
             LOG.warn("Failed to store case details", e);
             if (e.getCause() instanceof ConstraintViolationException) {
-                throw new CaseConcurrencyException(e.getMessage());
+                LOG.warn("ConstraintViolationException happen for UUID={}. Cause: {}",
+                    caseDetails.getReference(), ((ConstraintViolationException) e.getCause()).getConstraintName());
+                if (((ConstraintViolationException) e.getCause()).getConstraintName().equals(UNIQUE_INDEX_EXTERNAL_ID_CONSTRAINT)) {
```

Comment:
this is very specific to CMC index.. we might have different unique indexes as well (Ethos).

Reply:
Indeed, only CMC Jurisdiction has this index in place. It may not be needed to check the exact name of the constraint here to make your solution slightly more generic?

Reply:
Reversed this and now the code has nothing CMC specific.

Reply:
Looks good to me 👍 

Are we still going ahead with this change then? You mentioned that you would investigate the option of removing the retry on CaseConcurrencyExcpetion altogether...?